### PR TITLE
refactor!: remove `precision`, `decimal` inputs from `InputNumber` in favour of the `NumberFormat` directive

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
migrations and making directive standalone will be in a separate PR.

### CODE REVIEW

**Review by splincode — APPROVED** [review]

**Comment by vladimirpotekhin** [comment]
migrations and making directive standalone will be in a separate PR.

File: projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    >
+        Rounding
+    </ng-template>
+    <ng-template
+        documentationPropertyMode="input"
+        documentationPropertyName="decimalLimit"
+        documentationPropertyType="number"
+        [(documentationPropertyValue)]="documentedComponent.decimalLimit"
+    >
+        A number of digits after comma ( Infinity for an untouched decimal part)
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html:49]:
```suggestion
        A number of digits after comma ( <code>Infinity</code> for an untouched decimal part)
```

File: projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    </ng-template>
+    <ng-template
+        documentationPropertyMode="input"
+        documentationPropertyName="zeroPadding"
+        documentationPropertyType="boolean"
+        [(documentationPropertyValue)]="documentedComponent.zeroPadding"
+    >
+        Zero padding
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html:57]:
```suggestion
        If number of digits after <code>decimalSeparator</code> is always equal to the <code>decimalLimit</code>
```

File: projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    </ng-template>
+    <ng-template
+        documentationPropertyMode="input"
+        documentationPropertyName="thousandSeparator"
+        documentationPropertyType="string"
+        [(documentationPropertyValue)]="documentedComponent.thousandSeparator"
+    >
+        Thousand separator
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html:65]:
```suggestion
        Symbol for separating thousands
```

File: projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    </ng-template>
+    <ng-template
+        documentationPropertyMode="input"
+        documentationPropertyName="decimalSeparator"
+        documentationPropertyType="string"
+        [(documentationPropertyValue)]="documentedComponent.decimalSeparator"
+    >
+        Decimal separator
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html:73]:
```suggestion
        Symbol for separating fraction
```

File: projects/demo/src/modules/components/input-range/input-range.component.ts [hunk 5]
```diff
@@ -18,9 +19,10 @@ import {ABSTRACT_PROPS_ACCESSOR} from '../abstract/inherited-documentation/abstr
             provide: ABSTRACT_PROPS_ACCESSOR,
             useExisting: forwardRef(() => ExampleTuiInputRangeComponent),
         },
+        tuiDocExcludeProperties(['decimal', 'decimalLimit']),
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/input-range/input-range.component.ts:22]: `zeroPadding` do nothing for `InputRange`.
Should we exclude it too ?

https://taiga-ui--pr6997-number-format-direct-u23z2z12.web.app/components/input-range/API?zeroPadding=false&quantum=0.001

Comment (nsbarsukov) [reply projects/demo/src/modules/components/input-range/input-range.component.ts:22]: The same question for `InputSlider` page

File: projects/demo/src/modules/directives/number-format/examples/import/import-module.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+// ...
+
+@NgModule({
+  imports: [
+    // ...
+    TuiNumberFormatModule,
+  ],
+  // ...
+})
+export class MyModule {}
```
Comment (nsbarsukov) [inline projects/demo/src/modules/directives/number-format/examples/import/import-module.md:13]: Modern standalone example ?

File: projects/demo/src/modules/directives/number-format/number-format.template.html [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            <li>
+                <p>Add to the template:</p>
+
+                <tui-doc-code
+                    filename="my.component.html"
+                    [code]="exampleHtml"
+                ></tui-doc-code>
+            </li>
+        </ol>
+    </ng-template>
```
Comment (nsbarsukov) [inline projects/demo/src/modules/directives/number-format/number-format.template.html:46]: `<tui-setup />` ?

File: projects/kit/components/input-number/input-number.component.ts [hunk 8]
```diff
@@ -318,6 +313,12 @@ export class TuiInputNumberComponent
         return maskitoParseNumber(this.nativeValue, this.numberFormat.decimalSeparator);
     }
 
+    private get precision(): number {
+        return Number.isNaN(this.numberFormat.decimalLimit)
```
Comment (nsbarsukov) [inline projects/kit/components/input-number/input-number.component.ts:317]: When `decimalLimit` could be equal to `NaN` ?
Could we use `Infinity` instead `NaN`

Comment (vladimirpotekhin) [reply projects/kit/components/input-number/input-number.component.ts:317]: NaN is the default value. We use it to set different default values in different components

**Review by splincode — APPROVED** [review]

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 9]
```diff
@@ -343,6 +343,15 @@ test.describe('InputNumber', () => {
             await expect(example).toHaveScreenshot('27-input-number.png');
         });
 
+        test('Value 42,1 (decimal=pad) => 42,10', async ({page}) => {
+            await tuiGoto(page, '/components/input-number/API?precision=2&decimal=pad');
+
+            await input.fill('42,1');
+            await expect(input).toHaveJSProperty('selectionStart', 4);
+            await expect(input).toHaveJSProperty('selectionEnd', 4);
+            await expect(example).toHaveScreenshot('28-input-number.png');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:352]:
```suggestion
            await expect(input).toHaveValue('42,1');
            await input.blur();
            await expect(input).toHaveValue('42,10');
            await expect(example).toHaveScreenshot('28-input-number.png');
```

Comment (nsbarsukov) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:352]: (the same checks for the previous test)

File: projects/demo-playwright/tests/kit/input-number/input-number.spec.ts [hunk 10]
```diff
@@ -352,25 +361,25 @@ test.describe('InputNumber', () => {
...
 
-        test('Value 42 (decimal=never) => 42', async ({page}) => {
-            await tuiGoto(page, '/components/input-number/API?precision=2&decimal=never');
+        test('Value 42,1 (precision=0) => 42', async ({page}) => {
+            await tuiGoto(page, '/components/input-number/API?precision=0');
 
-            await input.fill('42');
-            await expect(example).toHaveScreenshot('29-input-number.png');
+            await input.fill('42,1');
+            await expect(example).toHaveScreenshot('30-input-number.png');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:371]:
```suggestion
            await expect(input).toHaveValue('42');
            await expect(example).toHaveScreenshot('30-input-number.png');
```

(additional security level 😄 )

Comment (nsbarsukov) [reply projects/demo-playwright/tests/kit/input-number/input-number.spec.ts:371]: Could you add it to other tests, please ?

File: projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+<tui-doc-documentation>
+    <ng-template
+        documentationPropertyMode="input"
+        documentationPropertyName="decimal"
+        documentationPropertyType="TuiDecimal"
+        [documentationPropertyValues]="documentedComponent.decimalVariants"
+        [(documentationPropertyValue)]="documentedComponent.decimal"
+    >
+        Show/hide decimal
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html:31]: Add this description, please

* `always` – number of digits after `decimalSeparator` is always equal to the `precision`.
* `pad` – pads trailing zeroes up to precision **on blur**
* `non-zero` – drops trailing zeroes up to precision **on blur**

File: projects/kit/components/input-number/input-number.component.ts [hunk 12]
```diff
@@ -41,7 +41,7 @@ import {
     TUI_TEXTFIELD_SIZE,
     TUI_TEXTFIELD_WATCHED_CONTROLLER,
     tuiFormatNumber,
-    tuiGetFractionPartPadded,
+    // tuiGetFractionPartPadded,
```
Comment (nsbarsukov) [inline projects/kit/components/input-number/input-number.component.ts:44]: Delete ?

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html (hunk 11 above)
Comment (vladimirpotekhin) [reply projects/demo/src/modules/components/abstract/number-format-documentation/number-format-documentation.template.html]: added, but without `on Blur`, for non-input components/utilities

File: projects/core/constants/default-number-format.ts [hunk 13]
```diff
@@ -2,10 +2,9 @@ import {CHAR_NO_BREAK_SPACE} from '@taiga-ui/cdk';
 import type {TuiNumberFormatSettings} from '@taiga-ui/core/interfaces';
 
 export const TUI_DEFAULT_NUMBER_FORMAT: TuiNumberFormatSettings = {
-    decimalLimit: NaN,
+    precision: NaN,
```
[the remark this answers is not in the document]
Comment (vladimirpotekhin) [reply projects/core/constants/default-number-format.ts:5]: discussed, decided to keep `NaN`

**Pull request merged**